oauth2_http/javatests/com/google/auth/http/HttpCredentialsAdapterTest.java

```diff
@@ -45,4 +57,77 @@ public void initialize_populatesOAuth2Credentials() throws IOException {
     String authorizationHeader = requestHeaders.getAuthorization();
     assertEquals(authorizationHeader, expectedAuthorization);
   }
+
+  public MockLowLevelHttpResponse makeErrorResponse() {
+    MockLowLevelHttpResponse errorResponse = new MockLowLevelHttpResponse();
+    errorResponse.addHeader("custom_header", "value");
+    errorResponse.setStatusCode(HttpStatusCodes.STATUS_CODE_UNAUTHORIZED);
+    errorResponse.setContentType(Json.MEDIA_TYPE);
+    errorResponse.setContent("{\"error\":\"invalid credentials\"}");
+    return errorResponse;
+  }
+
+  public MockLowLevelHttpResponse makeSuccessResponse() {
+    MockLowLevelHttpResponse successResponse = new MockLowLevelHttpResponse();
+    successResponse.addHeader("custom_header", "value");
+    successResponse.setStatusCode(HttpStatusCodes.STATUS_CODE_OK);
+    successResponse.setContentType(Json.MEDIA_TYPE);
+    successResponse.setContent(SUCCESS_CONTENT);
+    return successResponse;
+  }
+
+  @Test
+  public void initialize_populatesOAuth2Credentials_handle401() throws IOException {
+    final String accessToken = "1/MkSJoj1xsli0AccessToken_NKPY2";
+    final String accessToken2 = "2/MkSJoj1xsli0AccessToken_NKPY2";
+    final String expectedAuthorization = AuthHttpConstants.BEARER_PREFIX + accessToken;
+
+    final MockTokenServerTransport tokenServerTransport = new MockTokenServerTransport();
+    tokenServerTransport.addClient(CLIENT_ID, CLIENT_SECRET);
+    tokenServerTransport.addRefreshToken(REFRESH_TOKEN, accessToken);
+
+    OAuth2Credentials credentials = new UserCredentials(
+        CLIENT_ID, CLIENT_SECRET, REFRESH_TOKEN, tokenServerTransport);
+    credentials.refresh();
+    HttpCredentialsAdapter adapter = new HttpCredentialsAdapter(credentials);
+
+    HttpTransport primaryHttpTransport = new MockHttpTransport() {
+        @Override
```

Comment:
As discussed offline, this is separate from serving tokens - I'll move it into a separate mock class distinct from MockTokenServerTransport.